extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1671,12 +1685,20 @@ export class AmpA4A extends AMP.BaseElement {
     const analyticsVars = Object.assign(
         {'time': Math.round(this.getNow_())},
         this.getA4aAnalyticsVars(analyticsEvent));
-    triggerAnalyticsEvent(this.element, analyticsEvent, analyticsVars);
+    if (this.win.document == analyticsElement.ownerDocument) {
+      // Check if the amp-analytics is still attached to the document before
+      // firing the trigger.
+      // Note that this happens in tests when promise chains continue executing
+      // after the DOM has been cleaned up. I don't expect such scenarios to
+      // occur in practice.
+      triggerAnalyticsEvent(analyticsElement, analyticsEvent, analyticsVars);
+    }
```

Comment:
Maybe add a dev().warn here if the documents don't match.

Reply:
Done